qubes/vm/qubesvm.py

```diff
@@ -216,6 +216,59 @@ def _setter_kbd_layout(self, prop, value):
     return value
 
 
+def _default_active_bootmode(self):
+    """
+    Return the template's default bootmode for AppVMs if possible, otherwise
+    return a blank string. May reset the template's AppVM default bootmode
+    property if the boot mode it specifies does not exist.
+    """
+    subject = self
+    while hasattr(subject, "template"):
+        if hasattr(subject.template, "appvm_default_bootmode"):
+            bootmode_value = subject.template.appvm_default_bootmode
+            kernelopts = subject.features.check_with_template(
+                f"boot-mode.{bootmode_value}.kernelopts", None
+            )
+            if kernelopts is None:
+                subject.template.appvm_default_bootmode = qubes.property.DEFAULT
+            else:
+                return bootmode_value
+        subject = subject.template
+    return ""
```

Comment:
Your specification says the default boot mode is `base`. But the code in core_features.py assumes it's called `default`. Neither one is empty string... (personally I like `default` more)

Reply:
The default boot mode being named was only important for the fully feature-based implementation. It's easier IMO to deal with an empty string than a special value since we're using properties now.

Reply:
What about user-visible name for the default boot mode? IIUC setting that is supported, right? With default name being empty that means feature request `boot-mode..name`, which feels wrong.

Reply:
I'm not sure that a user-visible name for the default is possible or desirable, since the frontend PR is using `utils.initialize_widget_for_property` with `allow_default=True`, so the default entry always looks like `default (internal-bootmode-id)`. I also don't know if it's really necessary, though it would be *nice* to have. I'll try to look into it more.

If there is a way to get a user-friendly boot mode name in the default entry, then `boot-mode.default.name` can be assigned to it. I agree, `boot-mode..name` feels wrong.

Reply:
Resolved, now using `default`.